### DISCUSSION TITLE
When the to_image method is called, the PDF file is alaways occupied and cannot be closed

### DISCUSSION THREAD

**Discussion by uyoufu** [jsvine/pdfplumber#1126] [Ask for help with specific PDFs]
When the to_image method is called, the PDF file is alaways occupied and cannot be closed

How can I close the pdf file correctly?

Code is follow:

``` python
import pdfplumber
import os

pdf_path = "./test1.pdf"
pdf = pdfplumber.open(pdf_path)
try:
    for page_num, page in enumerate(pdf.pages):
        img = page.to_image()
        img.save(f"./test1_{page_num}.png")
finally:
    pdf.close()
    os.remove(pdf_path)

```

**Comment by jsvine** [comment]
What version of `pdfplumber` are you using? I believe the file-closing has been handled automatically since `0.10.4`, released in Feburary: https://github.com/jsvine/pdfplumber/blob/stable/CHANGELOG.md#0104---2024-02-10 / https://github.com/jsvine/pdfplumber/pull/1090

**Reply by uyoufu** [reply]
Thank you very much, I'm using 0.10.3. When I upgraded to 0.11.0, it was back to normal
